atlasdb-client/src/test/java/com/palantir/atlasdb/table/description/SchemaTest.java

```diff
@@ -181,7 +197,186 @@ public void testLongIndexNameLengthFailsCassandra() throws IOException {
+    }
+
+    @Test
+    public void indexTablesWithCleanupTasksHaveArbitraryCleanupMetadata() {
+        Schema schema = getSchemaWithSimpleTestTable();
+
+        addTestIndexDefinition(schema);
+        schema.addCleanupTask(TEST_INDEX_NAME, () -> (tx, cells) -> false);
+
+        assertCleanupMetadataInSchemaSatisfies(schema, INDEX_TABLE_REF, ARBITRARY_CLEANUP_METADATA_ASSERTION);
+    }
+
+    @Test
+    public void indexTablesWithoutCleanupTasksHaveNullCleanupMetadataEvenIfBaseTableHasCleanupTasks() {
+        Schema schema = getSchemaWithSimpleTestTable();
+        schema.addCleanupTask(TEST_TABLE_NAME, () -> (tx, cells) -> false);
+
+        addTestIndexDefinition(schema);
+
+        assertCleanupMetadataInSchemaSatisfies(schema, INDEX_TABLE_REF, NULL_CLEANUP_METADATA_ASSERTION);
+    }
+
+    @Test
+    public void streamStoreMetadataGeneratedCorrectlyForSimpleStreamStore() {
+        Schema schema = new Schema("Table", TEST_PACKAGE, Namespace.EMPTY_NAMESPACE);
+        String shortName = "f";
+        String longName = "floccinaucinihilipilification";
+
+        schema.addStreamStoreDefinition(
+                new StreamStoreDefinitionBuilder(shortName, longName, ValueType.VAR_LONG)
+                        .build());
+
+        assertCleanupMetadataCorrectForStreamStore(schema, shortName, 0, ValueType.VAR_LONG);
+    }
+
+    @Test
+    public void streamStoreMetadataGeneratedCorrectlyForStreamStoreWithHashFirstRowComponent() {
+        Schema schema = new Schema("Table", TEST_PACKAGE, Namespace.EMPTY_NAMESPACE);
+        String shortName = "a";
+        String longName = "antidisestablishmentarianism";
+
+        schema.addStreamStoreDefinition(
+                new StreamStoreDefinitionBuilder(shortName, longName, ValueType.FIXED_LONG)
+                        .hashFirstRowComponent()
+                        .build());
+
+        assertCleanupMetadataCorrectForStreamStore(schema, shortName, 1, ValueType.FIXED_LONG);
+    }
+
+    @Test
+    public void streamStoreMetadataGeneratedCorrectlyForStreamStoreWithHashRowComponents() {
+        Schema schema = new Schema("Table", TEST_PACKAGE, Namespace.EMPTY_NAMESPACE);
+        String shortName = "p";
+        String longName = "pneumonoultramicroscopicsilicovolcanoconiosis";
+
+        schema.addStreamStoreDefinition(
+                new StreamStoreDefinitionBuilder(shortName, longName, ValueType.FIXED_LONG_LITTLE_ENDIAN)
+                        .hashRowComponents()
+                        .build());
+
+        assertCleanupMetadataCorrectForStreamStore(schema, shortName, 2, ValueType.FIXED_LONG_LITTLE_ENDIAN);
+    }
+
+    @Test
+    public void streamStoreTablesSupportAdditionalCleanupTasks() {
+        Schema schema = new Schema("Table", TEST_PACKAGE, Namespace.EMPTY_NAMESPACE);
+        String shortName = "l";
+        String longName = "llanfairpwllgwyngyllgogerychwyrndrobwllllantysiliogogogoch";
```

Comment:
Now you're just showing off like that weather forecast guy

Reply:
It is a _long_ name after all... unfortunately I struggle pronouncing this one, unlike [this guy](https://www.youtube.com/watch?v=fHxO0UdpoxM)